embedded/sql/stmt.go

```diff
@@ -238,56 +236,69 @@ type CreateTableStmt struct {
 	table       string
 	ifNotExists bool
 	colsSpec    []*ColSpec
-	pk          string
+	pkColNames  []string
 }
 
 func (stmt *CreateTableStmt) inferParameters(e *Engine, implicitDB *Database, params map[string]SQLValueType) error {
 	return nil
 }
 
 func (stmt *CreateTableStmt) compileUsing(e *Engine, implicitDB *Database, params map[string]interface{}) (summary *TxSummary, err error) {
-	summary = newTxSummary()
-	summary.db = implicitDB
-
 	if implicitDB == nil {
 		return nil, ErrNoDatabaseSelected
 	}
 
+	summary = newTxSummary(implicitDB)
+
 	if stmt.ifNotExists && implicitDB.ExistTable(stmt.table) {
 		return summary, nil
 	}
 
-	table, err := implicitDB.newTable(stmt.table, stmt.colsSpec, stmt.pk)
+	table, err := implicitDB.newTable(stmt.table, stmt.colsSpec)
 	if err != nil {
 		return nil, err
 	}
 
-	e.catalog.mutated = true
+	createIndexStmt := &CreateIndexStmt{unique: true, table: table.name, cols: stmt.pkColNames}
+	indexSummary, err := createIndexStmt.compileUsing(e, implicitDB, params)
+	if err != nil {
+		return nil, err
+	}
 
-	for colID, col := range table.ColsByID() {
-		v := make([]byte, 1+len(col.colName))
+	err = summary.add(indexSummary)
+	if err != nil {
+		return nil, err
+	}
 
-		if col.autoIncrement && (col.colName != table.pk.colName || table.pk.colType != IntegerType) {
-			return nil, ErrLimitedAutoIncrement
-		}
+	for _, col := range table.Cols() {
+		//{auto_incremental | nullable}{maxLen}{colNAME})
+		v := make([]byte, 1+4+len(col.colName))
 
 		if col.autoIncrement {
+			if col.id != table.primaryIndex.cols[0].id {
```

Comment:
In only checks the first column, is that intentional?

I was able to run this SQL statement:

```CREATE TABLE test1(id INTEGER AUTO_INCREMENT, val INTEGER, PRIMARY KEY (id, val))```

Reply:
good. I think it's enough to limit auto_increment to single column PKs. I'll add the validation